electron/generators/diograph-generator.js

```diff
@@ -0,0 +1,68 @@
+const { isEmpty, promiseAllReduce } = require('../lib/utils')
```

Comment:
Are these generators only part of the "read folder tool" or are they used elsewhere?

Should we already have a concept of "tool" at least in the code-folder level? That all the code related to one tool would be in their own folder. In this case `electron/tools/read-folder/generators/diograph-generator`